build.groovy

```diff
@@ -427,7 +429,11 @@ def createBuildList() {
 	// check if full build
 	if (props.fullBuild) {
 		println "** --fullBuild option selected. $action all programs for application ${props.application}"
-		buildSet = buildUtils.createFullBuildList()
+		if (repositoryClient) {
+			buildSet = buildUtils.createFullBuildList() }
+		else {
+			println "*! Full build requires a repository client connection to a DBB web application"
+		}	
```

Comment:
You are changing the current behavior here. Any reason to force providing a repository connection in case of a full build?
I very often perform a full build without repository connection just to test that my whole application can compile.

Reply:
Interesting use case, I have not thought about this case so far. Always interesting to see the different usages.

Doesn't your use cause an issue when zAppBuild would like to update the collection? See line 517 in build.groovy

Reply:
Perhaps this is behavior that @drbruce-git has intentionally implemented in its design.

Reply:
I had always assumed that full build required a repository client so that it can store collections and build results for subsequent incremental builds.